### ISSUE TITLE
feat-wish: "Console" or "Log" view to see notification history

### ISSUE THREAD

**Issue by ozra** [howl-editor/howl#324]
Preferably filterable on "errors".

This would be a great help reviewing what's happening, since the notifications can quickly pass by at times.

**Comment by shalabhc** [comment]
I actually had a command in my init.moon that does this. Here it is (but it needs a patch to log.moon as well - to record the timestamp):

```moonscript
howl.command.register
  name: 'show-log-messages'
  description: 'Show all log messages in a buffer'
  handler: ->
    buf = howl.ui.ActionBuffer!
    buf.title = '[Log Messages]'
    levelmap = {
      'error':   'error   '
      'info':    'info    '
      'warning': 'warning '
    }
    current_date = os.date('%Y-%m-%d', os.time!)
    for entry in *log.entries
      entry_date = os.date('%Y-%m-%d', entry.timestamp)
      entry_time = os.date('%H:%M:%S', entry.timestamp)
      if entry_date != current_date
        entry_time = entry_date .. ' ' .. entry_time
      level = entry.level
      buf\append markup.howl("<comment>#{entry_time}</> <#{level}>#{levelmap[level]}</> #{entry.message}\n")
    buf.read_only = true
    howl.app\add_buffer buf
```

Once it's in a buffer, you can just use grep to filter for error if you want. It could use some more smarts (e.g. collapsing repeating messages into one, limiting the total number of entries shown etc.)

**Comment by ozra** [comment]
Thanks @shalabhc! I just stripped off the timestamp for now, and added `markup = howl.ui.markup` to the top (if any one else cuts n paste...)

That should be in the core!! PR :-)

**Comment by shalabhc** [comment]
Feel free to submit a PR! FWIW, I find it a bit less useful without the time ("did this happen in the last minute? yesterday?")

**Comment by ozra** [comment]
Yes, timestamp is essential of course, but for the temporary hack however, I'm just happy that I get an overview over the last few messages than not :-)

I'll see if I find the time to bake it together. Trying to keep this friday dry and do some contribs :) Working on #327 right now, it takes some time for me, still new to the code landscape and language (Moonscript turned out to miss some semantics I assumed it had)

**Comment by shalabhc** [comment]
This is now implemented as `open-journal` since 0.6.
